### PR TITLE
[11.4] Add parameters to the list of project repository tags

### PR DESCRIPTION
**Pull request by bbujisic**
Implementing the optional parameters from https://docs.gitlab.com/ee/api/tags.html#list-project-repository-tags

### CODE REVIEW

File: src/Api/Tags.php [hunk 1]
```diff
@@ -82,4 +91,19 @@ public function updateRelease($project_id, string $tag_name, array $params = [])
     {
         return $this->put($this->getProjectPath($project_id, 'repository/tags/'.self::encodePath($tag_name).'/release'), $params);
     }
+
+    /**
+     * @return OptionsResolver
+     */
+    protected function createOptionsResolver(): OptionsResolver
```
Comment (glensc) [inline src/Api/Tags.php]: it is not a very good idea to overwrite method with different purpose. another developer, already acquainted with this codebase, may not have idea that in this specific class $this-> createOptionsResolver() behaves differently than in all other classes.

indeed, they are able to discover it if they start troubleshooting, and find this difference, already by that time they are annoyed by the lost time and their motivation goes down.

suggestion: do not override the method, use local `$resolver` in the `all()` method. like rest of the similar code in this project. the `$resolver` does not have common options with any other method in this class.

Comment (glensc) [reply src/Api/Tags.php]: if you insist on a separate method, perhaps `createOptionsResolverForMethodAll`, the first suggestion to use a name like `createOptionsResolverAll` will only create additional confusion as it's not clear is for "all" methods or just the fact that the method name is `all` :D

also, use `private`, not `protected` by default in new methods, if there's no known reason to use wider visibility.

Comment (bbujisic) [reply src/Api/Tags.php]: I copied the solution from existing classes and existing `all()` methods. See `Issues::createOptionsResolver()` or `AbstractApi::createOptionsResolver()` for examples.

However, I agree with your argument that the override doesn't make sense as it won't be used anywhere else in the class and doesn't contribute the readability of the code.

The code is amended.

**Comment by GrahamCampbell** [comment]
Thank you, both.

**Comment by GrahamCampbell** [comment]
You can install this code using the version constraint `^11.4@dev`.

**Pull request merged**